application/src/main/java/org/togetherjava/tjbot/commands/reminder/RemindRoutine.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    private static RestAction<ReminderRoute> computeReminderRoute(@NotNull JDA jda, long guildId,
+            long channelId, long authorId) {
+        // If guild and channel can still be found, send there
+        Guild guild = jda.getGuildById(guildId);
+        if (guild != null) {
+            TextChannel channel = guild.getTextChannelById(channelId);
+            if (channel != null) {
+                return createGuildReminderRoute(jda, authorId, channel);
+            }
+        }
+
+        // Otherwise, attempt to DM the user directly
+        return createDmReminderRoute(jda, authorId);
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createGuildReminderRoute(@NotNull JDA jda,
+            long authorId, @NotNull TextChannel channel) {
+        return jda.retrieveUserById(authorId)
+            .onErrorMap(error -> null)
+            .map(author -> new ReminderRoute(channel, author,
+                    author == null ? null : author.getAsMention()));
+    }
+
+    private static @NotNull RestAction<ReminderRoute> createDmReminderRoute(@NotNull JDA jda,
+            long authorId) {
+        return jda.openPrivateChannelById(authorId)
+            .onErrorMap(error -> null)
+            .map(channel -> channel == null ? new ReminderRoute(null, null, null)
+                    : new ReminderRoute(channel, channel.getUser(),
+                            "(Sending your reminder directly, because I was unable to"
+                                    + " locate the original channel you wanted it to be send to)"));
+    }
+
+    private static void sendReminderViaRoute(@NotNull RestAction<ReminderRoute> routeAction,
+            long id, @NotNull CharSequence content, @NotNull TemporalAccessor createdAt) {
+        routeAction.flatMap(route -> {
+            if (route.isUndeliverable()) {
+                throw new IllegalStateException("Route is not deliverable");
+            }
+
+            MessageEmbed embed = createReminderEmbed(content, createdAt, route.target());
+            if (route.description() == null) {
+                return route.channel().sendMessageEmbeds(embed);
+            }
+            return route.channel().sendMessage(route.description()).setEmbeds(embed);
+        }).queue(message -> {
+        }, failure -> logger.warn(
+                "Failed to send a reminder (id '{}'), skipping it. This can be due to a network issue,"
+                        + " but also happen if the bot disconnected from the target guild and the"
+                        + " user has disabled DMs or has been deleted.",
+                id));
+    }
+
+    private static @NotNull MessageEmbed createReminderEmbed(@NotNull CharSequence content,
+            @NotNull TemporalAccessor createdAt, @Nullable User author) {
+        String authorName = author == null ? "Unknown user" : author.getAsTag();
+        String authorIconUrl = author == null ? null : author.getAvatarUrl();
```

Comment:
This method only runs if ReminderRoute#isUndeliverable is false, so why is User nullable?

And use User#getEffectiveAvatarUrl, this one doesn't work when no avatar has been set